witness/golang/api/http.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import "context"
+
+const (
+	// HTTPGetCheckpoint is the path of the URL to get a checkpoint.
+	HTTPGetCheckpoint = "witness/v0/get-checkpoint"
```

Comment:
A thought - up to you whether or not to adopt it, but RESTful versions of these URLs might be `/witness/v0/logs/<logid>/checkpoint` and `/witness/v0/logs/<logid>/update` (or even GET `/witness/v0/logs/<logid>/checkpoint` to fetch the latest, and POST to the same URL to update :D).
This would remove LogID from the GET params and POST body.

Reply:
Good idea!  I've changed the URLs to include the logid as `/witness/v0/{checkpoint,update}/<logid>`, although I do agree that what you have is more elegant :).

Reply:
Following the RESTful _thing_, if you stuck with `/witness/v0/logs/<logid>/...` style you could then have the witness return a list of all log IDs it knows about to a GET request at `/witness/v0/logs` and have it all kinda fit together...

(some rabbithole/bedtime reading here: https://restfulapi.net/rest-api-design-tutorial-with-example/ and https://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#restful :D)

Reply:
+1

Reply:
Okay, done!  Concretely I added the extra endpoint for getting the list of logs, which means I:
- Added an `HTTPGetLogs` endpoint in api/http.go
- Added a `GetLogs` function in witness.go and tested it in witness_test.go
- Added a `getLogs` function in server.go
I also changed the format and handlers for the other endpoints, in line with Al's initial suggestion.